src/trans.c

```diff
@@ -1243,7 +1211,7 @@ Obj FuncIMAGE_SET_TRANS_INT(Obj self, Obj f, Obj n)
         }
         SHRINK_PLIST(new, (Int)rank);
         SET_LEN_PLIST(new, (Int)rank);
-        SORT_PLIST_CYC(new);
+        SortPlistByMasterPointer(new);
         RetypeBag(new, T_PLIST_CYC_SSORT);
```

Comment:
Another reminder to self: This line is buggy, it is missing a `+ IMMUTABLE`. For that matter, one could simply create the bag with the right TNUM, and avoid the `RetypeBag` call here (assuming `SortPlistByMasterPointer` would not call RESET_FILTER)